domain/src/main/java/org/oppia/android/domain/util/StateRetriever.kt

```diff
@@ -171,7 +172,18 @@ class StateRetriever @Inject constructor(
           answerGroupJson.optJSONObject("rule_types_to_inputs"), interactionId
         )
       )
-      .build()
+
+    if (answerGroupJson.optJSONObject("tagged_skill_misconception_id") != null) {
```

Comment:
Suggest factoring the retrieved JSON object into a local val to avoid having to re-grab it. I think that will simplify the implementation here some.

Reply:
Fixed!